src/XMonad/Core.hs

```diff
@@ -345,6 +369,11 @@ class (Show (layout a), Typeable layout) => LayoutClass layout a where
     description :: layout a -> String
     description      = show
 
+instance (PureLayout layout a) => LayoutClass (Pure layout) a where
+    pureLayout (MkPure l) = pureLayout' l
+    pureMessage (MkPure l) = fmap MkPure . pureMessage' l
```

Comment:
Better to use `coerce` than `fmap MkPure` (may require type ascriptions or applications).